doc/source/serve/scaling-and-resource-allocation.md

```diff
@@ -0,0 +1,100 @@
+# Scaling and Resource Allocation
+
+This page should help you:
+
+- Scale your deployments horizontally by specifying a number of replicas
+- Scale up and down automatically to react to changing traffic
+- Allocate hardware resources (CPUs, GPUs, etc) for each deployment
+
+(scaling-out-a-deployment)=
+
+## Scaling horizontally with `num_replicas`
+
+Each deployment consists of one or more [replicas](serve-architecture#high-level-view).
+The number of replicas is specified by the `num_replicas` field in the deployment options.
+By default, `num_replicas` is 1.
```

Comment:
also, please specify:
- Each replica runs in a separate process (Ray actor).
- Replicas can run across multiple nodes in a Ray cluster.
- Each replica will have its own resources. We explain more below, but you can say by default each gets a single CPU and then link to the below section for more details.